add-ons/updating/tb78/convert-legacy.md

```diff
@@ -0,0 +1,146 @@
+# Converting legacy add-ons
+
+This guide is intended to help developers to port their add-ons to Thunderbird 78. While the intention of the guide is to simplify the porting process, porting legacy add-ons is not a straightforward process and requires a lot of time and some degree of development experience.
+
+
+## Dropping the legacy key and adding new entry points
+
+The technical conversion of a legacy add-on is ridiculously easy: just drop the `legacy` key from the `manifest.json` file.
+
+Now your add-on should install in current versions of Thunderbird without issues, but do nothing. After all, Thunderbird no longer calls anything within your add-on. To fix that, you need one or more entry points.
+
+There are multiple ways to get an add-on to load \(documented [in the WebExtensions course on MDN](https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/Anatomy_of_a_WebExtension)\), but for a Thunderbird add-on the most common option will be adding a *background script*:
+``` // manifest.json
+"background": {
+  "scripts": ["background-script.js"]
+}
+```
+Adding this secton to `manifest.json` will cause the file `background-script.js` to be loaded and evaluated by Thunderbird. For bootstrapped add-ons, the existing bootstrap script itself is a good starting point for a background script – for overlayed add-ons it may be reasonable to start with an empty script and convert overlays using the guidelines below, gradually building up the background script.
+
+Contrary to a bootstrap script in a bootstrapped add-on, background scripts will *not* get evaluated in a browser context. Instead they are added to a HTML document ("background page") living in a content process, which has access to [WebExtension APIs](https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/API) and [MailExtension APIs](https://thunderbird-webextensions.readthedocs.io/en/latest/index.html) throgh the `browser` and `messenger` globals. Some APIs may require specific permissions or settings to be specified within `manifest.json`, and some WebExtension APIs may be disabled or missing if they are not relevant to Thunderbird.
+
+Any interaction with Thunderbird must occur through these APIs. Whenever code needs to be added to the background script, you need to make sure to migrate calls to XPCOM or other native Thunderbird features to these APIs.
+
+
+## Experiment-ing with new APIs
+
+While the Thunderbird team plans to add more APIs with upcoming releases, the current set of APIs will not be sufficient to port most add-ons. To work around this limitation, add-ons can introduce their own, additional APIs as so-called [*experiments*](https://thunderbird-webextensions.readthedocs.io/en/latest/how-to/experiments.html). **Any feature that was available in previous versions of Thunderbird remains available in Thunderbird 78 when using experiments.**
+
+Experiments consist of three parts:
```

Comment:
I assume your comment is about the three parts, and not about the introduction statement? The intro statement is imho essential (what are experiments, when do I need them, what can they do). I was torn on the second part, but included it because of the 'deep' things you mention in the next comment. So this is best discussed below.

Reply:
Yes, I clicked on line 29 to add the comment, I refer to the part in lines 29+. Do not know why github decided to display it like that.